### PR TITLE
yesod-bin: Improve the 'use stack' message

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: yesod-bin/main.hs [hunk 1]
```diff
@@ -144,6 +144,17 @@ main = do
                 hPutStrLn stderr "Instead, please just run 'stack test'"
                 exitFailure
 
+    initErrorMsg = do
+        mapM_ putStrLn
+            [ "The init command has been removed."
+            , "Please use 'stack new <project name> <template>' instead where the"
+            , "available templates can be found by running 'stack templates'. For"
+            , "a Yesod based application you should probably choose one of the"
+            , "pre-canned yesod templates."
```
Comment (MaxGabriel) [inline yesod-bin/main.hs]: "Yesod" is capitalized on L152 but lowercase on L153

**Comment by MaxGabriel** [comment]
LGTM

**Comment by erikd** [comment]
Fixed capitalisation.

**Comment by MaxGabriel** [comment]
> because ghc-8.0 prints call stack info for `error` calls.

Very happy about the reason for this PR :)

**Comment by erikd** [comment]
Yeah, ghc-8.0 has a bunch of great new features like:

* Call stacks.
* `StrictData` and `Strict` language pragma.
* Warnings for redundant type class constraints.
* Reduced compile times (especially for the vector package).
* Applicative Do

and more.

**Pull request merged**